### PR TITLE
Added LinkPreview dependencies

### PR DESCRIPTION
**Pull request by AdamDotCom**

### CODE REVIEW

File: src/components/button-link/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    };
+
+    if (this.props.openInNewTab) {
+      linkProps.target = '_blank';
+      linkProps.rel = 'noreferrer noopener';
+    }
+
+    return (
+      <a {...linkProps}>
+        <div className='button-link__label'>
```
Comment (mrval5) [inline src/components/button-link/index.tsx]: not sure why we need `div` inside `a`

File: src/components/button-link/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (!/^http[s]?:\/\/(.*)$/.test(url)) {
+      protocol = 'https://';
+    }
+
+    return `${protocol}${url}`;
+  }
+
+  renderLink() {
+    const linkProps: any = {
```
Comment (mrval5) [inline src/components/button-link/index.tsx]: Can we change `any` to proper type

File: src/components/button-link/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export interface Properties {
+  url: string;
+  className?: string;
+  openInNewTab?: boolean;
+}
+
+export class ButtonLink extends React.Component<Properties, undefined> {
```
Comment (mrval5) [inline src/components/button-link/index.tsx]: maybe `undefined` can be confusing, can we remove it

File: src/components/video-player/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    const height = Math.max(boundingHeight, boundingWidth / ratio);
+
+    this.setState({
+      width: width + 'px',
+      height: height + 'px',
+    });
+  }
+
+  get hasDimensions() {
+    return this.state.width !== '0px' && this.state.height !== '0px';
```
Comment (mrval5) [inline src/components/video-player/index.tsx]: can we make `0px` a constant to reuse it

**Review by mrval5 — CHANGES_REQUESTED** [review]

File: src/components/button-link/index.tsx (hunk 1 above)
Comment (AdamDotCom) [reply src/components/button-link/index.tsx]: @mrval5, here's the original component. This PR is basically a verbatim copy: https://github.com/m3m3n70/zero-web/blob/development/src/components/button-link/index.tsx

**Pull request merged**